common/djangoapps/student/helpers.py

```diff
@@ -270,6 +260,29 @@ def get_next_url_for_login_page(request):
     return redirect_to
 
 
+def get_redirect_to(request):
+    """
+    Determine the redirect url and return if safe
+    :argument
+        request: request object
+
+    :returns: redirect url if safe else None
+    """
+    redirect_to = request.GET.get('next')
+    header_accept = request.META.get('HTTP_ACCEPT', '')
+
+    # if we get a redirect parameter, make sure it's safe and redirecting to web page.
+    # If it's not, drop the parameter.
+    if redirect_to and (not http.is_safe_url(redirect_to) or 'text/html' not in header_accept):
```

Comment:
I think you should still add the check to ensure the request URL do not contain `settings.STATIC_URL`.